### ISSUE TITLE
[Bug]: FilterableMultiSelect does not support `home` and `end` keys

### ISSUE THREAD

**Issue by tay1orjones** [carbon-design-system/carbon#13433]
### Package

@carbon/react

### Browser

Chrome

### Package version

v1.25.0

### React version

v17

### Description

In inputs, the `home` or `end` keys or `fn + left arrow` or `fn + right arrow` keys should move the cursor in the input to the beginning of the text or the end of the text (respectively).

FilterableMultiSelect does not support this for some reason. `cmd + left/right arrow` does work, but it should be in addition to the keys mentioned above.

This originally [surfaced in slack](https://ibm-studios.slack.com/archives/C2K6RFJ1G/p1680181817262399)

### Reproduction/example

react.carbondesignsystem.com

### Steps to reproduce

Go to FilterableMultiSelect, input text, press `home` or `end` key or `fn + left arrow` or `fn + right arrow` - note that these keys do not move the cursor at all, but they should.
